File: 448_find_all_numbers_disappeared_in_an_array/Solution1/Solution.java
import java.util.*;

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> result = new ArrayList<>();
        if (nums == null || nums.length == 0) return result;
        int[] tmpArr = new int[nums.length];

        for (int num : nums) tmpArr[num - 1]++;
        for (int i = 0; i < tmpArr.length; i++) {
            if (tmpArr[i] == 0) result.add(i + 1);
        }

        return result;
    }

    public static void main(String[] args) {
        int[] nums = {4, 3, 2, 7, 8, 2, 3, 1};
        System.out.println(new Solution().findDisappearedNumbers(nums));
    }
}

File: 448_find_all_numbers_disappeared_in_an_array/Solution1/Solution.py
class Solution:
    def findDisappearedNumbers(self, nums: list) -> list:
        if not nums: return []
        tmp_arr = [0] * len(nums)
        for num in nums: tmp_arr[num - 1] += 1
        return [i + 1 for i, num in enumerate(tmp_arr) if num == 0]

if __name__ == '__main__':
    nums = [4, 3, 2, 7, 8, 2, 3, 1]
    print(Solution().findDisappearedNumbers(nums))

